guard/src/migrate/parser.rs

```diff
@@ -147,13 +153,52 @@ impl Display for Rule {
     }
 }
 
+#[derive(Eq, PartialEq, Debug, Clone, Hash)]
+pub(crate) struct Dedup_anchor {
+    pub type_name: TypeName,
+    pub property: String,
+}
+
 #[derive(PartialEq, Debug, Clone, Hash)]
 pub(crate) struct Clause {
     pub(crate) rules: Vec<Rule>
 }
 impl Display for Clause {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let result: Vec<String> = self.rules.clone().into_iter().map(|rule| format!("{}", rule)).collect();
+
+        let mut property_to_rules = HashMap::new();
+        let cloned_rules = self.rules.clone();
+        let mut result= Vec::new();
+
+        for rule in cloned_rules.iter() {
+            match rule {
+                Rule::Conditional(conditional) => { result.push(format!("{}", rule)) },
+                Rule::Basic(base) => {
+                    property_to_rules.entry(Dedup_anchor{
+                        type_name: TypeName{type_name: format!("{}",base.type_name)},
+                        property: format!("{}", base.property_comparison.property_path)})
+                        .or_insert_with(Vec::new).push(rule);
+                }
+            }
+        }
+
+        for (anchor,rules) in property_to_rules.into_iter() {
+            if rules.len() == 1 {
+                result.push(format!("{}", rules[0]))
+            } else {
+                let anchor_value = anchor.type_name;
+                let mut result_for_key= Vec::new();
+                for(rule) in rules.iter() {
+                    match rule {
+                        Rule::Conditional(conditional) => {},
```

Comment:
This should never occur based on the code above from 176 - 181. Just use the `if-let` form for these use cases, like `if let Rule::Basic(basic) = rule {...}`